### DISCUSSION TITLE
@livewireScripts or @livewireScriptsConfig in app.blade.php Query

### DISCUSSION THREAD

**Discussion by robben07** [livewire/livewire#8973] [1. Help]
I wonder if someone can advise me on whether I need to use @livewireScripts or @livewireScriptsConfig in app.blade.php.

When I use @livewireScriptsConfig in the file (see below), the literal string appears on the screen. However, when I use @livewireScripts, it doesn’t display. I believe I should be using @livewireScriptsConfig.

Thanks in advance for your help!

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="csrf-token" content="{{ csrf_token() }}">

    <title>TestApp</title>
    @vite(['resources/css/app.css', 'resources/js/app.js'])
</head>
<body class="bg-gray-100">
    <div class="container mx-auto px-4 py-8">
        {{ $slot }}
    </div>

    @livewireScriptsConfig
</body>
</html>
```

**Comment by EdgarsJoja** [comment]
I think it entirely depends on what are your needs in this case.

1) I don't need (for now) much of any additional JS libraries, where loading order would matter so I don't include anything at all. Neither `@livewireScripts`, nor `@livewireScriptConfig` injection is required for Livewire setup.

2) If you need to control "where" and "when" to inject scripts & Alpine, use `@livewireScripts`.

3) If you need even more control, of defining exactly how JS is packaged & loaded, use `@livewireScriptConfig`.

At least this is my understanding. All in all, unless you hit some specific problem, I would go with (1) initially. (2) & (3) require more fiddling around, and I would advise to avoid them initially unless actually needed.
Careful re-read of installation docs also wouldn't hurt imho: https://livewire.laravel.com/docs/installation

EDIT: I'm talking about V3 here

EDIT 2: Correct naming of `@livewireScriptConfig`, as pointed out by jeff.

**Comment by robben07** [comment]
Many thanks for your reply, it's much appreciated.

**Reply by jeffchown** [reply]
@robben07 Also, if you do use it, it is `@livewireScriptConfig` (not `Scripts` - see https://livewire.laravel.com/docs/installation#manually-bundling-livewire-and-alpine)

This explains why it was appearing as a literal string on the screen when you tried it.

**Reply by robben07** [reply]
Thanks Jeff for confirming.
